src/config/calling_convention.cpp

```diff
@@ -166,5 +184,10 @@ std::ostream& operator<<(std::ostream &out, const CallingConvention& cc)
 	return out;
 }
 
+CallingConventionID CallingConvention::operator=(const CallingConventionID& cc)
```

Comment:
C++ tip: Unless there is a specific reason for returning a value from an assignment operator, [it is generally recommended](http://www.icu-project.org/docs/papers/cpp_report/the_anatomy_of_the_assignment_operator.html) to return a reference:
> What does the function return? An assignment operator can return anything it wants, but the standard C and C++ assignment operators return a reference to the left-hand operand. This allows you to chain assignments together like so:
>
> ```cpp
> x = y = z = 3;
> ```
>
> Unless you have a really good reason, you want to follow this convention. Returning a reference to the right-hand operand or a value (i.e., another whole TFoo) would both still allow the simple chain described above to work, but have subtle differences in semantics from the way the standard operators do it that would come out in more complicated expressions involving assignment. Returning a value also forces unnecessary trips through the object's copy constructor, costing you in performance.
>
> So the outer shell of a properly-written assignment operator would look like this:
>
> ```cpp
> TFoo&
> TFoo::operator=(const TFoo& that)
> {
>     // copy the state...
>     return *this;
> }
> ```
>
> (By the way, note that the return statement is "return *this", rather than "return this". That's because this is a pointer to TFoo. You have to dereference it to get a reference to TFoo. Of course, if you forget this, the compiler will remind you.)

Reply:
Also, an assignment operator for class `X` should generally return a reference to `X` and not some other type (e.g. a member as in your case).

Reply:
Thank you for the tip. I will incorporate the recommendation and update commit.